riptide-stream/src/main/java/org/zalando/riptide/stream/StreamSpliterator.java

```diff
@@ -47,6 +47,7 @@ public boolean tryAdvance(Consumer<? super T> action) {
         try {
             JsonToken token = parser.nextToken();
             if (token == null) {
+                parser.close();
```

Comment:
I do not understand. We only close automatically when reaching the end. When we never reach the end, I do not now how I should close. The Spliterator itself is not closeable and will not be close by the stream support. I agree, we are something missing here and that's why I added this line here.